### DISCUSSION TITLE
windows Apache + PHP 8.1  Unable to find library 'libvips-42.dll'

### DISCUSSION THREAD

**Discussion by flpxra** [libvips/php-vips#185] [Q&A]
Hi,
I'm having trouble running libvips under Windows.
Of course I have FFI enabled, I have the correct path pointed to libvips, and it still doesn't work. 
@jcupitt could you help me on this topic?

Error:
`Fatal error: Uncaught Jcupitt\Vips\Exception: Unable to find library 'libvips-42.dll' in any of ['D:\wamp64\www\Tests\libvips-phpbinding/lib64/', 'D:\wamp64\www\Tests\libvips-phpbinding/lib/']. Make sure that you've installed libvips and that 'libvips-42.dll' is on your system's library search path.`

My sample php code:
```
<?php
putenv('VIPSHOME='.dirname(__FILE__));

require __DIR__ . '/vendor/autoload.php';
use Jcupitt\Vips;

$image = Vips\Image::thumbnail('somefile.jpg', 128);
$image->writeToFile('tiny.jpg');
```

**Comment by jcupitt** [comment]
Hi @flexphperia,

Could it be a 32/64 bit mismatch? You'll need to use 32-bit everywhere (php and libvips) or 64-bit everywhere.

Did you see this issue? https://github.com/libvips/php-vips/issues/183#issuecomment-1402149080 There are detailed install notes for php-vips on windows. You could try those (they should work), then check that install against your main one for differences.

**Comment by flpxra** [comment]
I'm using 64 bit everywhere, confirmed. Also checked issue that you mention - no luck.

**Reply by flpxra** [reply]
I have PHP 8.1: compiler Visual C++ 2019, x64 installed as Apache handler but I think it does not matter.

**Reply by jcupitt** [reply]
Oh, what exactly is in the `lib` directory you have on your path? Could you copy-paste the directory contents?

It could also be a clash with another package you have installed. What else is on your path?

**Comment by jcupitt** [comment]
Could you try following the steps I list in that issue exactly? That should work, so then you just need to find the differences between that working install and the one that's failing.
